From 17cd2aeadeea2393786ac244c263cd9ceb98d9cd Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Sun, 23 Jul 2017 13:36:16 +0300
Subject: [PATCH 1153/1345] fix: net: mvpp2x: fix mask in parser default
 multicast entry

commit  94dfa0fac15e6ec44fa19c3ffb5394e3aa549262 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- before this patch, only multicast traffic with explicit values of
  0x01 or 0x33 in the MAC address most significant byte were
  allowed (but i.e. 0x11 was dropped)
- mask in multicast entry is set to match the multicast value
- i.e. for ipv4 multicast default entry in parser
  the value is set to 1 and the mask was set to 0xff.
  Any multicast traffic entering the device with a
  multicast address with first MAC address byte different
  than the value 0x1 was not hit by the rule
- after this fix, all multicast traffic will pass through
  the system when multicast traffic is enabled
- this patch affects multicast behaviour

Change-Id: I999918d3b7d2525c404512fcf4202a2feb3b373c
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41987
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
index 5851c3c..d2c2a57 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
@@ -991,7 +991,7 @@ void mv_pp2x_prs_mac_multi_set(struct mv_pp2x_hw *hw, int port, int index,
 					   MVPP2_PRS_RI_L2_CAST_MASK);
 
 		/* Update tcam entry data first byte */
-		mv_pp2x_prs_tcam_data_byte_set(&pe, 0, da_mc, 0xff);
+		mv_pp2x_prs_tcam_data_byte_set(&pe, 0, da_mc, da_mc);
 
 		/* Shift to ethertype */
 		mv_pp2x_prs_sram_shift_set(&pe, 2 * ETH_ALEN,
-- 
1.7.9.5

